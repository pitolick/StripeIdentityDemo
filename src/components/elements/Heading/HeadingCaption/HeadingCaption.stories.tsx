import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { HeadingCaption } from './HeadingCaption'

const meta: Meta<typeof HeadingCaption> = {
  title: 'elements/Heading/HeadingCaption',
  component: HeadingCaption,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof HeadingCaption>

/**
 * デフォルト
 */
export const Default: Story = {
  args: {
    children: '見出し',
  },
}

/**
 * h1
 */
export const H1: Story = {
  args: {
    children: '見出し',
    tag: 'h1',
  },
}

/**
 * h2
 */
export const H2: Story = {
  args: {
    children: '見出し',
    tag: 'h2',
  },
}

/**
 * h3
 */
export const H3: Story = {
  args: {
    children: '見出し',
    tag: 'h3',
  },
}

/**
 * h4
 */
export const H4: Story = {
  args: {
    children: '見出し',
    tag: 'h4',
  },
}

/**
 * h5
 */
export const H5: Story = {
  args: {
    children: '見出し',
    tag: 'h5',
  },
}

/**
 * h6
 */
export const H6: Story = {
  args: {
    children: '見出し',
    tag: 'h6',
  },
}

/**
 * カスタムクラス
 */
export const CustomClass: Story = {
  args: {
    children: '見出し',
    className: 'text-red',
  },
}

/**
 * 子要素
 */
export const Children: Story = {
  args: {
    children: (
      <>
        <span className="text-4xl">これは見出しです。</span>
        <br />
        <span className="text-xl">これはサブタイトルです。</span>
      </>
    ),
  },
}
