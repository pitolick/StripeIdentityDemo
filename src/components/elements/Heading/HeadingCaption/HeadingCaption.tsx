/**
 * 装飾付きの見出し
 * @module HeadingCaption
 * @example
 *  <HeadingCaption>見出し</HeadingCaption>
 */
import React from 'react'

/**
 * 装飾付きの見出しコンポーネントのプロパティ
 */
interface HeadingCaptionProps {
  /**
   * 子要素
   */
  children: React.ReactNode
  /**
   * タグ名
   */
  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  /**
   * class名
   */
  className?: React.HTMLAttributes<HTMLHeadingElement>['className']
}

/**
 * 装飾付きの見出しコンポーネント
 */
export const HeadingCaption: React.FC<HeadingCaptionProps> = ({ children, tag: CustomTag = 'h2', className }) => {
  return (
    <>
      <CustomTag className={`${className} relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] `}>
        <span className="relative">{children}</span>
      </CustomTag>
    </>
  )
}
