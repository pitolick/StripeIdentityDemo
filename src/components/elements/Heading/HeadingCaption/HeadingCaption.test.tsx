import React from 'react'
import { render } from '@testing-library/react'
import { HeadingCaption } from './HeadingCaption'

describe('HeadingCaptionコンポーネント', () => {
  it('子要素が正しく表示されること', () => {
    const { container } = render(<HeadingCaption>見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h1タグが正しく設定されること', () => {
    const { container } = render(<HeadingCaption tag="h1">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h2タグが正しく設定されること', () => {
    const { container } = render(<HeadingCaption tag="h2">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h3タグが正しく設定されること', () => {
    const { container } = render(<HeadingCaption tag="h3">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h4タグが正しく設定されること', () => {
    const { container } = render(<HeadingCaption tag="h4">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h5タグが正しく設定されること', () => {
    const { container } = render(<HeadingCaption tag="h5">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h6タグが正しく設定されること', () => {
    const { container } = render(<HeadingCaption tag="h6">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('class名が正しく設定されること', () => {
    const { container } = render(<HeadingCaption className="text-red">見出し</HeadingCaption>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
