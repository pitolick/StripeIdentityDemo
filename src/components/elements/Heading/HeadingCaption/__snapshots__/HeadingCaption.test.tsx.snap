// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingCaptionコンポーネント class名が正しく設定されること 1`] = `
<h2
  class="text-red relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h2>
`;

exports[`HeadingCaptionコンポーネント h1タグが正しく設定されること 1`] = `
<h1
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h1>
`;

exports[`HeadingCaptionコンポーネント h2タグが正しく設定されること 1`] = `
<h2
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h2>
`;

exports[`HeadingCaptionコンポーネント h3タグが正しく設定されること 1`] = `
<h3
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h3>
`;

exports[`HeadingCaptionコンポーネント h4タグが正しく設定されること 1`] = `
<h4
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h4>
`;

exports[`HeadingCaptionコンポーネント h5タグが正しく設定されること 1`] = `
<h5
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h5>
`;

exports[`HeadingCaptionコンポーネント h6タグが正しく設定されること 1`] = `
<h6
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h6>
`;

exports[`HeadingCaptionコンポーネント 子要素が正しく表示されること 1`] = `
<h2
  class="undefined relative ps-4 font-bold before:absolute before:inset-y-0 before:left-0 before:block before:w-2 before:bg-gray-500 before:content-[''] "
>
  <span
    class="relative"
  >
    見出し
  </span>
</h2>
`;
